BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -147,6 +147,184 @@ public static Vector3 GetAirToGroundTarget(Vector3 targetPosition, Vector3 targe
             return targetPosition + (targetVelocity * leadTime);
         }
 
+        public static Vector3 GetAirToAirLoftTarget(Vector3 targetPosition, Vector3 targetVelocity,
+            Vector3 targetAcceleration, Vessel missileVessel, float targetAlt, float maxAltitude,
+            float rangeFactor, float altComp, float velComp, float loftAngle, float termAngle,
+            float termDist, ref int loftState, out float timeToImpact, out float targetDistance,
+            float minSpeed = 200)
+        {
+
+            Vector3 velDirection = missileVessel.Velocity().normalized;
+
+            targetDistance = Vector3.Distance(targetPosition, missileVessel.transform.position);
+
+            float currSpeed = Mathf.Max((float)missileVessel.srfSpeed, minSpeed);
+            Vector3 currVel = currSpeed * velDirection;
+
+            //Vector3 Rdir = (targetPosition - missileVessel.transform.position).normalized;
+            //float rDot = Vector3.Dot(targetVelocity - currVel, Rdir);
+
+            float leadTime = (float)(1 / ((targetVelocity - currVel).magnitude / targetDistance));
+            //float leadTime = (targetDistance / rDot);
+
+            timeToImpact = leadTime;
+            leadTime = Mathf.Clamp(leadTime, 0f, 8f);
+
+            // If loft is not terminal
+            if ((targetDistance > termDist) && (loftState < 3))
+            {
+                if (BDArmorySettings.DEBUG_MISSILES) Debug.Log("[BDArmory.MissileLauncher]: Lofting");
+
+                // Get up direction
+                Vector3 upDirection = VectorUtils.GetUpDirection(missileVessel.CoM);
+
+                // Use the gun aim-assist logic to determine ballistic angle (assuming no drag)
+                Vector3 bulletRelativePosition, bulletRelativeVelocity, bulletAcceleration, bulletRelativeAcceleration, targetPredictedPosition, bulletDropOffset, lastVelDirection, ballisticTarget, targetHorVel, targetCompVel;
+                
+                var firePosition = missileVessel.transform.position + (currSpeed * velDirection) * Time.fixedDeltaTime; // Bullets are initially placed up to 1 frame ahead (iTime). Not offsetting by part vel gives the correct initial placement.
```

Comment:
The initial placement correction wouldn't apply here as that's specific to how bullets are implemented to properly space out bullets firing faster than the framerate.

Reply:
Removed the corrections.